app/models/option_node.rb

```diff
@@ -245,6 +245,12 @@ def arrange_as_rows(hash = nil, parent_path = [])
     end
   end
 
+  def preferred_name_translations(path)
+    path.map do |path|
```

Comment:
Using `path` as loop var is confusing. For short loops I usually use the initials of the obj class. So in this case you could just use `o`.